data/outport/txWithFee.go

```diff
@@ -6,43 +6,61 @@ import (
 	"github.com/ElrondNetwork/elrond-go-core/data"
 )
 
-type transactionHandlerWithGasAndFee struct {
-	data.TransactionHandler
+// FeeInfo holds information about the fee and gas used
+type FeeInfo struct {
+	GasUsed        uint64
+	Fee            *big.Int
+	InitialPaidFee *big.Int
+}
 
-	gasUsed uint64
-	fee     *big.Int
+// TransactionHandlerWithGasAndFee holds a data.TransactionHandler and information about fee and gas used
+type TransactionHandlerWithGasAndFee struct {
+	data.TransactionHandler
```

Comment:
very very risky business having a struct exported along with exported fields. Why? I can easily panic your code by typing this:
```GO
t := &outport.TransactionHandlerWithGasAndFee(....)
t.TransactionHandler = nil
t.<any method that uses TransactionHandler>
```

Please convert this struct to unexported and use unexported field. If you still want to embed TransactionHandler into your struct, you might create an unexported type alias, something like:
```GO
type transactionHandler = data.TransactionHandler

type TransactionHandlerWithGasAndFee struct{
    transactionHandler
}

```

Reply:
If I do this the `data.TransactionHandler` will be ignored but the marshaller because is an unexported field

Reply:
ok, agree to remain as it is if this will remain as a DTO (that might implement an interface) **without any business logic**